src/code/z_lights.c

```diff
@@ -1,15 +1,83 @@
 #include "global.h"
 #include "assets/objects/gameplay_keep/gameplay_keep.h"
 
+// #define LIGHTS_DEBUG
+
+#if !(ENABLE_F3DEX3 || defined(F3DEX_GBI_PL))
+#error A microcode that supports point lighting is required
+#endif
+
+// Toggleable static for debugging, recommended for release is to use static
+#ifndef STATIC
+#if IS_DEBUG
+#define STATIC
+#else
+#define STATIC static
+#endif
+#endif
```

Comment:
Marking a function or data as static prevents it from appearing in the linker map file, which some don't like as they'd like to lookup addresses in the map file while debugging. This is a big reason why we've never moved to add static keywords to functions upstream.

Since debugging tools have gotten more sophisticated, like ares getting a gdb stub implementation this may start to no longer be a problem, but some may still appreciate the toggle.

Reply:
Got it. I think this PR should not introduce any new systems like this and stick with how the code is in the rest of the repo. If you would like to make a separate PR to globally introduce a static toggle, feel free.